src/test/java/com/github/checkstyle/UnifiedDiffReaderTest.java

```diff
@@ -0,0 +1,217 @@
+
+        final UnifiedDiffFile file2 = diff.getFiles().get(1);
+        assertEquals("diff -U0 old/f2 new/f2", file2.getDiffCommand());
+        assertEquals("old/f2", file2.getFromFile());
+        assertEquals("new/f2", file2.getToFile());
+        assertEquals(1, file2.getPatch().getDeltas().size());
+
+        assertNull(diff.getTail());
+    }
+
+    @Test
+    public void testMultiChangedFilesOnOnePatchTwo() throws Exception {
+        final UnifiedDiff diff = getUnifiedDiff("MultiChangedFilesOnOnePatch.txt");
+        assertEquals(2, diff.getFiles().size());
+
+        final UnifiedDiffFile file1 = diff.getFiles().get(0);
+        assertEquals("diff -u Origin/Test1.java Update/Test1.java",
+                file1.getDiffCommand());
+        assertEquals("Origin/Test1.java", file1.getFromFile());
+        assertEquals("Update/Test1.java", file1.getToFile());
+        assertEquals(1, file1.getPatch().getDeltas().size());
+
+        final UnifiedDiffFile file2 = diff.getFiles().get(1);
+        assertEquals("diff -u Origin/Test2.java Update/Test2.java",
+                file2.getDiffCommand());
+        assertEquals("Origin/Test2.java", file2.getFromFile());
+        assertEquals("Update/Test2.java", file2.getToFile());
+        assertEquals(1, file2.getPatch().getDeltas().size());
+
+        assertNull(diff.getTail());
+    }
+
+    @Test
+    public void testGitDiffPatch() throws Exception {
+        final UnifiedDiff diff = getUnifiedDiff("PatchFileFromDiffTools/GitDiffPatch.txt");
+        assertEquals(1, diff.getFiles().size());
+
+        final UnifiedDiffFile file1 = diff.getFiles().get(0);
+        assertEquals("diff --git a/pom.xml b/pom.xml", file1.getDiffCommand());
+        assertEquals("pom.xml", file1.getFromFile());
+        assertEquals("pom.xml", file1.getToFile());
+        assertEquals(1, file1.getPatch().getDeltas().size());
+
+        assertNull(diff.getTail());
+    }
+
+    @Test
+    public void testGitFormatPatchLinuxOne() throws Exception {
+        final UnifiedDiff diff =
+                getUnifiedDiff("PatchFileFromDiffTools/GitFormatPatchLinuxOne.txt");
+        assertEquals(1, diff.getFiles().size());
+
+        final UnifiedDiffFile file1 = diff.getFiles().get(0);
+        assertEquals("diff --git a/pom.xml b/pom.xml", file1.getDiffCommand());
+        assertEquals("pom.xml", file1.getFromFile());
+        assertEquals("pom.xml", file1.getToFile());
+        assertEquals(1, file1.getPatch().getDeltas().size());
+
+        assertEquals("2.7.4\n\n", diff.getTail());
+    }
+
+    @Test
+    @Ignore("error: com.github.difflib.unifieddiff.UnifiedDiffParserException: "
+            + "expected file start line not found "
+            + "until https://github.com/java-diff-utils/java-diff-utils/issues/84")
+    public void testGitFormatPatchLinuxTwo() throws Exception {
+        final UnifiedDiff diff =
+                getUnifiedDiff("PatchFileFromDiffTools/GitFormatPatchLinuxTwo.txt");
```

Comment:
Please provide details in this.
Des it mean that we will not be able load git patches that create files ?

Reply:
I mentioned here https://github.com/checkstyle/patch-filters/pull/16#discussion_r443110089